internal/exec/exec.go

```diff
@@ -51,7 +51,9 @@ func (r *Request) Execute(ctx context.Context, s *resolvable.Schema, op *query.O
 	}()
 
 	if err := ctx.Err(); err != nil {
-		return nil, []*errors.QueryError{errors.Errorf("%s", err)}
+		//If context has either been cancelled or timed out we still may want to return the features that have finished
+		// TODO properly attribute mark which features have timedout in the error field
```

Comment:
It seems like this PR somehow silences all errors if the context deadline is exceeded, even if those errors showed up before the deadline. Can you confirm if that's true or not? This might have been what you were mentioning to me the other day but I think I hadn't dug into the PR enough to understand the implications.

Reply:
I notice locally that if the context deadline is exceeded I only get that one error and if the context deadline is not exceeded I got all sorts of underlying errors.

Reply:
The behavior before this PR was to block until all resolvers have finished, aggregate the errors and then throw them away if the context deadline has been exceeded.

Currently if we get an error, and then another resolver makes the context deadline exceed then we will throw that error away.

Adding the ability to persist errors even in the face of a context deadline being exceeded seems like a reasonable scope for this PR.